src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenByUsernameAndPasswordConfidentialParameterBuilder.cs

```diff
@@ -0,0 +1,134 @@
+                   .WithScopes(scopes).WithUsername(username).WithPassword(password);
+        }
+
+        /// <summary>
+        /// Enables MSAL to read the federation metadata for a WS-Trust exchange from the provided input instead of acquiring it from an endpoint.
+        /// This is only applicable for managed ADFS accounts. See https://aka.ms/MsalFederationMetadata.
+        /// </summary>
+        /// <param name="federationMetadata">Federation metadata in the form of XML.</param>
+        /// <returns>The builder to chain the .With methods</returns>
+        public AcquireTokenByUsernameAndPasswordConfidentialParameterBuilder WithFederationMetadata(string federationMetadata)
+        {
+            Parameters.FederationMetadata = federationMetadata;
+            return this;
+        }
+
+        /// <summary>
+        ///  Modifies the token acquisition request so that the acquired token is a Proof-of-Possession token (PoP), rather than a Bearer token. 
+        ///  PoP tokens are similar to Bearer tokens, but are bound to the HTTP request and to a cryptographic key, which MSAL can manage on Windows.
+        ///  Note that only the host and path parts of the request URI will be bound.
+        ///  See https://aka.ms/msal-net-pop
+        /// </summary>
+        /// <param name="nonce">Nonce of the protected resource (RP) which will be published as part of the WWWAuthenticate header associated with a 401 HTTP response
+        /// or as part of the AuthorityInfo header associated with 200 response. Set it here to make it part of the Signed HTTP Request part of the POP token.</param>
+        /// <param name="httpMethod">The HTTP method ("GET", "POST" etc.) method that will be bound to the token. If set to null, the PoP token will not be bound to the method.
+        /// Corresponds to the "m" part of the a signed HTTP request.</param>
+        /// <param name="requestUri">The URI to bind the signed HTTP request to.</param>
+        /// <returns>The builder.</returns>
+        /// <remarks>
+        /// <list type="bullet">
+        /// <item><description>An Authentication header is automatically added to the request.</description></item>
+        /// <item><description>The PoP token is bound to the HTTP request, more specifically to the HTTP method (GET, POST, etc.) and to the Uri (path and query, but not query parameters).</description></item>
+        /// <item><description>Broker is required to use Proof-of-Possession on public clients.</description></item>
+        /// </list>
+        /// </remarks>
+#if iOS || ANDROID
+        [EditorBrowsable(EditorBrowsableState.Never)]
+#endif
+        public AcquireTokenByUsernameAndPasswordConfidentialParameterBuilder WithProofOfPossession(string nonce, HttpMethod httpMethod, Uri requestUri)
+        {
+            ClientApplicationBase.GuardMobileFrameworks();
+
+            if (!ServiceBundle.Config.IsBrokerEnabled)
```

Comment:
See this is where it starts to be strange. We're mixing PCA concepts in here. 

We could support ROPC + confidential client POP in reality, no?

Reply:
Yes, this broker code is only for PCA, will remove. I do think it is possible to perform ROPC + POP using the CCA POP implementation.

Reply:
can confirm the ROPC + POP using the CCA POP flow works

Reply:
I expected as much, POP is protocol agnostic.